Sources/MongoSwift/Operations/ListDatabasesOperation.swift

```diff
@@ -0,0 +1,91 @@
+import mongoc
+
+/// Options to use when listing available databases.
+internal struct ListDatabasesOptions: Encodable {
+    /// Optional `Document` specifying a filter that the listed databases must pass.
+    internal var filter: Document?
+
+    /// Optionally indicate whether only names should be returned.
+    /// This is internal and only used for implementation purposes. Users should use `listDatabaseNames` if they want
+    /// only names.
+    internal var nameOnly: Bool?
+
+    /// Convenience constructor for basic construction
+    internal init(filter: Document?, nameOnly: Bool?) {
+        self.filter = filter
+        self.nameOnly = nameOnly
+    }
+}
+
+/// A struct modeling the information returned from the `listDatabases` command about a single database.
+public struct DatabaseSpecification: Codable {
```

Comment:
I guess it does give us issues with forward compatibility, but in general we'd want to be encouraging our users to be using `Codable` over `Document` wherever possible, including in their own structs and what not. Providing APIs that return `Document` sort of goes against this principle. I personally would like to avoid a situation where we supply two ways to do something and prefer if users only use one of them; I'd rather just omit the other case.

Also, I think it's worth noting that the spec specifically calls out using a model as an acceptable return value and does not give the warning "format subject to change" that I have seen in other specs.

Finally, having two overloads only differ in return value makes type inference a little annoying for users.